packages/gestalt/src/boxTransforms.js

```diff
@@ -0,0 +1,648 @@
+// @flow strict
+import styles from './Box.css';
+import borders from './Borders.css';
+import colors from './Colors.css';
+import layout from './Layout.css';
+import whitespace from './boxWhitespace.css';
+import {
+  concat,
+  fromClassName,
+  fromInlineStyle,
+  identity,
+  mapClassName,
+  toProps,
+  type ToPropsOutput,
+  type Style,
+} from './style.js';
+import omit from './utils/omit.js';
+import {
+  union,
+  bind,
+  range,
+  toggle,
+  mapping,
+  rangeWithZero,
+  rangeWithoutZero,
+  type Functor,
+} from './transforms.js';
+import { getRoundingStyle } from './getRoundingClassName.js';
+import { type Indexable } from './zIndex.js';
+import {
+  type AlignContent,
+  type AlignItems,
+  type AlignSelf,
+  type BorderSize,
+  type Column,
+  type Color,
+  type DangerouslySetInlineStyle,
+  type Dimension,
+  type Display,
+  type Direction,
+  type Flex,
+  type JustifyContent,
+  type Margin,
+  type Opacity,
+  type Overflow,
+  type Padding,
+  type Position,
+  type ResponsiveProps,
+  type UserSelect,
+} from './boxTypes.js';
+
+/*
+
+# Transformers
+
+This is where the meat and the bones of Box's transforms are. You can read more about the DSL in `./transforms.js`, but basically they are a small declarative way of specifying how a property (i.e. `marginTop={4}`) gets turned into a CSS class (`marginTop4`).
+
+There's a little preamble here, but it culminates in a big object mapping the actual property names to the transformer values.
+
+*/
+
+const display: Functor<Display> = mapping({
```

Comment:
this…is unfortunate. Flow couldn't figure out the types for almost any of these when defined in the object literal, so I had to define them here to type the function outputs. why Flow was losing the types from the function definitions, I don't know.